plugin/storage/grpc/factory.go

```diff
@@ -98,7 +100,7 @@ func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger)
 		for _, opt := range opts {
 			clientOpts = append(clientOpts, configgrpc.WithGrpcDialOption(opt))
 		}
-		return f.config.ToClientConn(context.Background(), componenttest.NewNopHost(), telset, clientOpts...)
+		return f.config.ToClientConn(context.Background(), f.host, telset, clientOpts...)
```

Comment:
`f.host` could be nil if the factory is created via a different constructor. I am surprised no unit or integration test is catching that.

Reply:
Yes we need to add a nil check for `f.host` here and cover factory various factory initialization scenarios.